File: app/src/main/java/mitya/yahnc/ui/ShowStoriesFragment.java
package mitya.yahnc.ui;

import mitya.yahnc.network.StoryService;
import rx.Observable;
import rx.schedulers.Schedulers;

/**
 * Created by Mitya on 25.08.2016.
 */
public class ShowStoriesFragment extends RemoteStoryFragment {
    @Override
    protected Observable<Integer> getCurrentStories() {
        return StoryService.getInstance().getShowStories().
                flatMap(stories -> Observable.from(stories).subscribeOn(Schedulers.io()));
    }
}

File: app/src/main/java/mitya/yahnc/ui/ShowStoriesFragment.kt
package mitya.yahnc.ui

import io.reactivex.Observable
import io.reactivex.schedulers.Schedulers
import mitya.yahnc.network.HnService

/**
 * Created by Mitya on 25.08.2016.
 */
class ShowStoriesFragment : RemoteStoryFragment() {
    override fun getCurrentStories(): Observable<Int> {
        return HnService.getShowStories().flatMap { stories -> Observable.fromArray(*stories).subscribeOn(Schedulers.io()) }
    }
}
